### ISSUE TITLE
401 'Could not authenticate you.' with certain queries

### ISSUE THREAD

**Issue by chrisbrain** [JoeMayo/LinqToTwitter#84]
Hi,

If I change the search term in the test Search feature of the console app to the following:

```
 static async Task DoSearchAsync(TwitterContext twitterCtx)
        {
            string searchTerm = "\"LINQ to Twitter\" OR Linq2Twitter OR LinqToTwitter";
            searchTerm = "кот (";

```

I get an error:

(Inner Exception #0) LinqToTwitter.TwitterQueryException: Could not authenticate you. - Please visit the LINQ to Twitter FAQ (at the HelpLink) for help on resolving this error.
   at LinqToTwitter.TwitterErrorHandler.<HandleUnauthorizedAsync>d__4.MoveNext() in C:\Repos\LinqToTwitter\New\Libraries\LinqToTwitterPcl\LinqToTwitter\TwitterErrorHandler.cs:line 84
--- End of stack trace from previous location where exception was thrown ---

Actually, this repos the same behaviour I get in my own Twitter OAuth code (and even in another client I tested with). I was hoping LinqToTwitter may not exhibit this behaviour :)

I have actually posted on Stackoverflow a little more generally about what the issue is (as far as I can tell):
http://stackoverflow.com/questions/41696829/httpwebrequest-sometimes-incorrectly-decoding-parts-of-a-query-parameter

But essentially if you have search queries mixing certain characters (e.g. ( * :) with non latin characters (not sure that's exactly the right terminilogy) it seems to break the OAuth.

Interested if you have any thoughts on this or you know how to patch LinqToTwitter to work with these queries also.

**Comment by JoeMayo** [comment]
I've looked at this but haven't found a resolution yet. I believe it has something to do with character encoding. Also, the encoding of the query needs to match the encoding sent into OAuth. I believe the only place this happens is on the Search API, because I've been able to resolve these issues with other APIs. I haven't figured out what it might be about Search that's different.

**Comment by JoeMayo** [comment]
This is fixed and I'm planning on including it in a release early this week.

**Comment by JoeMayo** [comment]
This was fixed in [LINQ to Twitter v4.2.0](https://github.com/JoeMayo/LinqToTwitter/releases/tag/v4.2.0).

[@JoeMayo](https://twitter.com/JoeMayo)
